[Terence Chew] iP

chewterence opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

chewterence committed:
Create Duke chat bot with Greet, Echo and Exit

* add Main driver class
* add InputHandler class that processes inputs
* update Duke class to interface with main driver class

chewterence committed:
Add commands List and Add

* add command interface for future commands
* add Task class to encapsulate descriptions of tasks
* add TaskList class to encapsulate a list of tasks
* add ListCommand class that lists tasks from TaskList
* add AddCommand class that adds tasks to TaskList
* add ByeCommand class that exits the Duke chat bot

chewterence committed:
Add ability to mark tasks as done

* add DoneCommand that marks tasks as done
* update InputHandler to handle done command
* update Task class to encapsulate variables to track completion of tasks
* update TaskList class to handle modification of list of tasks

chewterence committed:
Add support for tracking Todo, Event and Deadline Tasks

* update AddCommand class to support three types of tasks
* add DeadlineTask class to encapsulate tasks with deadlines
* add EventTask class to encapsulate tasks with event durations
* add TodoTask class to encapsulate tasks without deadlines or durations
* update InputHandler to support addition of three types of tasks
* update Task class to support inheritance

chewterence committed:
Update text-ui-test to semi-automate testing of Duke chat bot

* add inputs and expected outputs of simple tests

chewterence committed:
Add ability to handle errors

* add DukeException class that represents errors
* update InputHandler to throw DukeExceptions when errors are encountered
* update Duke class to handle DukeExceptions

chewterence committed:
Remove busy waiting that causes input to not terminate

chewterence committed:
Add support for deleteing tasks from the list

* add DeleteCommand class that deletes tasks from taskList
* update InputHandler to support delete command
* update TaskList to support deleting of tasks

chewterence committed:
Add ability to save tasks in the hard disk automatically

* create StorageHandler class to encapsulate handling of reading and writing from files
* update AddCommand class to handle saving of files
* update Duke class to handle saving of files

chewterence committed:
Add ability to save tasks in the hard disk automatically

* create StorageHandler class to encapsulate handling of reading and writing from files
* update AddCommand class to handle saving of files
* update Duke class to handle saving of files

chewterence committed:
Merge branch 'branch-Level-7' of https://github.com/chewterence/ip into branch-Level-7

chewterence committed:
Extract AddCommand class into 3 classes

* separate AddToDo, AddDeadline and AddEvent classes.
* modify DeadlineTask class to encapsulate java LocalDate format
* update InputHandler class to handle parsing of inputs for java LocalDate format

chewterence committed:
Update DeadlineTask class to display LocalDate with format

chewterence committed:
Merge branch 'branch-Level-7'

chewterence committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/AddCommand.java

chewterence committed:
Rename InputHandler class and StorageHandler class

* rename InputHandler class to Parser to match project requirements
* rename StorageHandler class to Storage to match project requirements

chewterence committed:
Update storage for compatibility with LocalDate format

merging branch-Level-7 and branch-level-8 caused Storage class to not be able to handle java LocalDate format

* update DeadlineTask and EventTask to encapsulate serialization of java LocalDate format
* update Storage class to handle reading and writing to and from file with serialized java LocalDate format

chewterence committed:
Create Ui class to encapsulate output handling

chewterence committed:
Package classes

chewterence committed:
Add JUnit tests for automated testing

chewterence committed:
Add java documentation for command package

chewterence committed:
Restructure project directories

chewterence committed:
Add java documentation for task package

chewterence committed:
Update code to follow coding standard

chewterence committed:
Merge branch 'branch-A-JavaDoc'

chewterence committed:
Merge branch 'branch-A-CodingStandard'

chewterence committed:
Add support to find task by searching for keyword

chewterence committed:
Merge branch 'branch-Level-9'

chewterence committed:
Update code to follow styling conventions

chewterence committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

chewterence committed:
Update gradle configuration and add checkstyle

chewterence committed:
Add GUI using java fx

chewterence committed:
Merge branch 'branch-Level-10'

jxrrelo commented:
Hi Terence! I think it is commendable that most parts of your code is organized which makes it easy to read. It also follows closely with the Java coding standard with minor areas for improvement. The following are some feedback I have for you which might help you improve on your work quality.

jxrrelo commented:
Hi Terence! I think it is commendable that most parts of your code is organized which makes it easy to read. It also follows closely with the Java coding standard with minor areas for improvement. The following are some feedback I have for you which might help you improve on your work quality.

jxrrelo reviewed (commented):
Hi Terence! I think it is commendable that most parts of your code is organized which makes it easy to read. It also follows closely with the Java coding standard with minor areas for improvement. The following are some feedback I have for you which might help you improve on your work quality.

jxrrelo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,100 @@
+        }
+    }
+
+    private static AddDeadline processDeadline(String input) throws Exception {
```
Should the class name be Deadline instead of AddDeadline? The latter sounds more like what we would usually name a method. This also applies for some of the other classes you have.

jxrrelo commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,35 @@
+     * Returns a boolean value to signal the bot to exit.
+     * @return True if command signals bot to be terminated.
+     */
+    public boolean shouldExit() {
```
According to the Java coding standard for boolean variables/methods, could it be more appropriate to use isExit()?

jxrrelo commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class DoneCommand implements Command {
+
+    private final int id;
```
Should final variables should be in all capital letters, including an underscore if there are spaces present?

chewterence replied:
Thanks for your suggestion, but I believe you have made a mistake. Only static final (constants) are in screaming snake case, or what you refer to as in all capital letters.

jxrrelo commented on src/main/java/duke/command/HelpCommand.java:
```diff
@@ -0,0 +1,49 @@
+                + "\n'delete <duke.task number>'"
+                + "\nto delete duke.task from your list"
+                + "\n'bye'"
+                + "\nto exit the program";
```
I like how you organized this neatly because it is a long statement. Makes it really easy to read, keep it up!

chewterence replied:
Thanks jaryl i appreciate it

jxrrelo commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+     * Checks if the task is complete.
+     * @return A boolean value representing the completion status of the task.
+     */
+    public boolean isDone() {
```
Since this method is a getter used to fetch the value of isDone variable, should you be naming it as getIsDone() or any method name along those lines to avoid confusions between the class variable and class method?

chewterence committed:
Add more input validation

chewterence committed:
Add assertions

chewterence committed:
Merge pull request #1 from chewterence/branch-A-Assertions

Add assertions

chewterence committed:
Improve code quality

* delete unused imports in Duke class
* add enums and refactor methods in Parser class
* refactor methods in Storage class
* remove unused code in Ui class
* improve readability in TaskList class

chewterence committed:
Merge pull request #2 from chewterence/branch-A-CodeQuality

Improve code quality

jay9645 commented on src/main/java/Duke.java:
```diff
@@ -1,17 +1,24 @@
     private boolean isRunning;
     private TaskList taskList;
+    private final StorageHandler storageHandler;
+    private final String path = "ip/src/main/java/data/TaskListData.txt";
```
Since this string is a constant, maybe you can make it static and uppercase?

jay9645 commented on src/main/java/StorageHandler.java:
```diff
@@ -0,0 +1,99 @@
+                return new EventTask(description, taskID, status, lineContents[3]);
+            default:
+                throw new DukeException("Error in file reading: Unknown task type");
+        }
```
According to the module coding standard, there should not be indentation for case clauses

chewterence replied:
Thanks for pointing that out! i will take note and make the right changes 👍

jay9645 commented on src/main/java/InputHandler.java:
```diff
@@ -1,9 +1,14 @@
 
         String[] processedInput = input.split(" ");
         String command = processedInput[0];
+        String description = processDescription(processedInput);
```
I noticed that you often used the work "process..." when naming your functions. I think process is kinda vague and may confuse the reader. Maybe you can consider a change like convertDescriptionStringTo... instead of processDescription

jay9645 reviewed (commented):
Overall it was really well written and readable code! Your coding style feels very structured and I learnt a lot from it!

chewterence committed:
Fix bug that causes text to clip off

* replace DialogBox Label type with Text type
* update MainWindow and DialogBox fxml version

chewterence committed:
Add lambda and streams

chewterence committed:
Merge pull request #3 from chewterence/branch-A-Lambdas

Add lambda and streams

chewterence committed:
Merge remote-tracking branch 'origin/master'

chewterence committed:
Update code to follow checkstyle

* remove comments in Parser class
* remove unused imports in TaskList class

chewterence committed:
Add continuous integration

chewterence committed:
Rename superlinter.yml to gradle.yml

chewterence committed:
Update help command to match latest functionality

chewterence committed:
Merge branch 'branch-C-Help'

chewterence committed:
Add ability to detect clashing Deadline Tasks

chewterence committed:
Update TaskList class to follow styling guide

chewterence committed:
Merge pull request #5 from chewterence/branch-B-DetectAnomalies

Branch b detect anomalies

chewterence committed:
Fix bug that causes todo task printing incorrectly

bug causes the word "todo" to appear along with the description of the task when listed.

Update Parser class to remove the extra string of "todo".

chewterence committed:
Add more input validation

* add ability to check for 24hr time format
* update help command to display more information on command formatting

chewterence committed:
Rename Duke chat bot to Antonio chat bot

chewterence committed:
Update code style to follow checkstyle

chewterence committed:
Improve format of event task

event task now can be specified from a particular start date and time, to an end date and time.

* add InvalidFormatException class that is thrown when an invalid time format is parsed

chewterence committed:
Update Tasks to display 24 hour time format

chewterence committed:
Merge branch 'branch-ImproveDateTime'

chewterence committed:
Update java docs to follow styling conventions

chewterence committed:
Add user guide

chewterence committed:
Add representative screenshot

chewterence committed:
Update user guide

chewterence committed:
Update user guide to hyperlink to release page

chewterence committed:
Update README.md

chewterence committed:
Fix bug that allows input even after exit

* add user input disable to MainWindow class

chewterence committed:
Fix bug that allows an input of done and delete command index 0

* update validation regex in Parser class

chewterence committed:
Fix bug that causes find command to not search with whitespaces

* update Parser class to include whitespaces in search key string

chewterence committed:
Fix bug that causes todo task to be valid with empty description

* add input validation for todo tasks in Parser class

chewterence committed:
Refactor try catch block out of parse method

chewterence committed:
Fix bug causing deadline error message to show for event

chewterence committed:
Update deadline error message

chewterence committed:
Merge pull request #7 from chewterence/fix-peer-evaluation-bugs

Fix peer evaluation bugs

chewterence committed:
Update hyperlink to release in user guide

chewterence committed:
Update image source files
